internal/service/certificatesigningrequest.go

```diff
@@ -8,13 +8,58 @@ import (
 
 	api "github.com/flightctl/flightctl/api/v1alpha1"
 	"github.com/flightctl/flightctl/internal/api/server"
+	"github.com/flightctl/flightctl/internal/crypto"
 	"github.com/flightctl/flightctl/internal/flterrors"
 	"github.com/flightctl/flightctl/internal/service/common"
 	"github.com/flightctl/flightctl/internal/store"
 	"github.com/go-openapi/swag"
+	"github.com/google/uuid"
 	"k8s.io/apimachinery/pkg/labels"
 )
 
+const DefaultEnrollmentCertExpirySeconds int32 = 60 * 60 * 24 * 7 // 7 days
+
+func signApprovedCertificateSigningRequest(ca *crypto.CA, request *api.CertificateSigningRequest) error {
+	if request == nil {
+		return errors.New("signApprovedCertificateSigningRequest: CertificateSigningRequest is nil")
+	}
+
+	// NOTE: this currently only requires one approval; this will change in the future based on policy
+	if api.IsStatusConditionTrue(request.Status.Conditions, api.CertificateSigningRequestDenied) {
+		return fmt.Errorf("signApprovedCertificateSigningRequest: certificate signing request has been denied and cannot be signed")
+	}
+	if !api.IsStatusConditionTrue(request.Status.Conditions, api.CertificateSigningRequestApproved) {
+		return fmt.Errorf("signApprovedCertificateSigningRequest: certificate signing request is not approved and cannot be signed")
+	}
+
+	csr, err := crypto.ParseCSR(request.Spec.Request)
+	if err != nil {
+		return fmt.Errorf("signApprovedCertificateSigningRequest: error parsing CSR: %w", err)
+	}
+
+	if err := csr.CheckSignature(); err != nil {
+		return err
+	}
+
+	// as we don't have a specific device yet, generate a UUID to represent the future device
+	uuid := uuid.NewString()
```

Comment:
We should use the csr Subject CommonName if already provided as part of the CSR.